packages/ossa/src/utils/index.ts

```diff
@@ -0,0 +1,30 @@
+const cache: Set<string> = new Set();
+/**
+ * 当使用了了过期属性时显示警告
+ * @param moduleName 模块名称
+ * @param newPropName 新的属性名
+ * @param oldPropName 旧的属性名
+ */
+export function warnDeprecatedProp(moduleName: string, newPropName: string, oldPropName: string): void {
+    const msg = `在 ${moduleName} 组件当中的 '${oldPropName}' 属性将在未来的版本中废弃，请使用 '${newPropName}' 代替!`;
+    if(cache.has(msg)) return;
+    cache.add(msg);
+    console.warn(`%c🚨 ${msg}`, 'font-weight: 800;');
+}
+/**
+ * 安全地使用过期属性
+ * @param newProp 新属性值
+ * @param oldProp 旧属性值
+ * @param waringInfo 当用户传入了过期的属性值时，如果如果传入了相关警告信息，将会自动在开发时显示警告
+ * @returns 
+ */
+export function deprecatedProp<T = unknown, P = T>(newProp: T, oldProp: P, waringInfo?: {
+    newPropName: string;
+    oldPropName: string;
+    moduleName: string;
+}): T | P {
+    if(oldProp && waringInfo && waringInfo.oldPropName && process.env.NODE_ENV === "development") {
```

Comment:
目前仅适用于默认为`false`的属性，通过判断oldProp是否传值来进行判断，是不是可以让这个方法变得更通用些？
@kiner-tang 

```suggestion
    if(typeof oldProp !== 'undefined' && waringInfo && waringInfo.oldPropName && process.env.NODE_ENV === "development") {
```

Reply:
是的，这里确实应该这么处理比较好